include/deal.II/base/symmetric_tensor.h

```diff
@@ -1584,10 +1590,37 @@ namespace internal
     };
 
 
+
+    /**
+     * Replacement of std::swap with device support
+     * for float and double (std::swap is host only)
+     */
+    template <class NumberType>
+    DEAL_II_HOST_DEVICE void
+    swap(NumberType &x, NumberType &y) noexcept
+    {
+      if constexpr (std::is_same_v<NumberType, double> ||
+                    std::is_same_v<NumberType, float> ||
+                    std::is_same_v<NumberType, unsigned int>)
```

Comment:
What about `int`? `char`? Perhaps use `std::is_integral_v || std::is_floating_point_v`? (The union is equivalent to `std::is_arithmetic_v`: https://en.cppreference.com/w/cpp/types/is_arithmetic.html)

Reply:
This helper function is used to get SymmetricTensor on device. I don't see a reason to support any other type than what I have already. I can change it if you feel strongly about it. At that point it might make sense to move this into a more general place (not sure where).

Reply:
Hm, thinking about this now, then I'm not happy that we put this function into the entire deal.II namespace under a generic name if it is only used in `SymmetricTensor`. If that's the only place where you use it, then let's hide the function in some internal namespace with a name that clearly indicates that it's only to be used in a specific context. If you want it to be generally usable in other places as well, then it should perhaps go into `utilities.h` (namespace `Utilities`) but then it also ought to support all datatypes that one can actually use on a device.

Reply:
Ah, I missed that it's in the internal namespace. Good to go then!